modules/fx/arrow-fx/src/main/kotlin/arrow/fx/internal/IOBracket.kt

```diff
@@ -1,23 +1,24 @@
 package arrow.fx.internal
 
-import arrow.core.Either
 import arrow.core.nonFatalOrThrow
 import arrow.fx.IO
-import arrow.fx.ForIO
 import arrow.fx.IOConnection
 import arrow.fx.IOFrame
 import arrow.fx.IOOf
+import arrow.fx.IOResult
 import arrow.fx.IORunLoop
-import arrow.fx.typeclasses.CancelToken
-import arrow.fx.typeclasses.ExitCase
+import arrow.fx.fix
+import arrow.fx.flatMap
+import arrow.fx.handleErrorWith
+import arrow.fx.typeclasses.ExitCase2
 import kotlinx.atomicfu.atomic
 
 internal object IOBracket {
 
   /**
    * Implementation for `IO.bracketCase`.
    */
-  operator fun <A, B> invoke(acquire: IOOf<A>, release: (A, ExitCase<Throwable>) -> IOOf<Unit>, use: (A) -> IOOf<B>): IO<Nothing, B> =
+  operator fun <E, E2 : E, A, B> invoke(acquire: IOOf<E, A>, release: (A, ExitCase2<E>) -> IOOf<E2, Unit>, use: (A) -> IOOf<E2, B>): IO<E2, B> =
```

Comment:
I'm confuesed here, is it because E is invariant but we need it in a contravariant position?